test-data/unit/check-newsyntax.test

```diff
@@ -151,3 +151,9 @@ v = 1
 reveal_type(f'{v}') # N: Revealed type is "builtins.str"
 reveal_type(f'{1}') # N: Revealed type is "builtins.str"
 [builtins fixtures/f_string.pyi]
+
+[case testFeatureVersionSuggestion]
+# flags: --python-version 3.99
+this is what future python looks like public static void main String[] args await goto exit
+[out]
+main:2: error: invalid syntax; maybe you need to use a newer version of Python to run mypy?
```

Comment:
I agree with @KotlinIsland, I think we need to show versions to users and explain what is going on with more details. Because it is confusing indeed.

```suggestion
main:2: error: invalid syntax; maybe you need to use a newer version of Python to run mypy?
python3.10 cannot parse syntax from python 3.99
```

Reply:
What about we update the `ast` module in older versions of python to be able to parse newer python? 

Could also make it an installable package.

Reply:
I maintain to many `ast` hacks to agree 🙂 
Example: https://github.com/wemake-services/wemake-python-styleguide/tree/master/wemake_python_styleguide/compat

Reply:
Yeah, agreed that's a nice to have. But I don't have time to spend wiring up the test infra just to allow a version specific error message for this.
If this is an improvement over the status quo, let's merge, otherwise I'll close this and someone else can look into shipping it with a better error message :-)

Reply:
That's basically what typed-ast was, and it was a bit of a pain to maintain